recipes/full_finetune_distributed.py

```diff
@@ -590,7 +582,7 @@ def recipe_main(cfg: DictConfig) -> None:
             "Distributed finetune recipe should be run via a distributed launcher."
             "If using tune CLI, please specify --nnodes 1 and --nproc_per_node [num_gpus]"
         )
-
+    os.environ["TORCH_NCCL_AVOID_RECORD_STREAMS"] = "1"
```

Comment:
Can you explain what this does?

Reply:
Oh yeah this is a good question. This was originally used to save memory when we were using FSDP with `sync_module_states=True` (see #843). I suspect it shouldn't be needed with our current FSDP2 usage but can also run a quick sanity check here. @weifengpy can you confirm whether we hit `dist._broadcast_coalesced` in any part of `fully_shard`?

Reply:
We can remove setting TORCH_NCCL_AVOID_RECORD_STREAMS in FSDP2 because we get rid of recordstream. `dist._broadcast_coalesced ` was from state dict loading for FSDP1 and it's gone as well